src/Servers/Kestrel/Kestrel/src/WebHostBuilderKestrelExtensions.cs

```diff
@@ -42,6 +42,8 @@ public static IWebHostBuilder UseKestrel(this IWebHostBuilder hostBuilder)
             // Don't override an already-configured transport
             services.TryAddSingleton<IConnectionListenerFactory, SocketTransportFactory>();
 
+            services.AddNamedPipes();
```

Comment:
Do we want to add the named pipes transport by default? Could this hurt trimming? @eerhardt

Reply:
I don't expect name pipes would include much. The types are thin wrappers around Win32 APIs on Windows and System.Net.Socket on Linux. However, it would be good to measure.

If the named pipes transport factory isn't registered by default, then we'd need to make sure that calling `ListenNamePipe` has a clear error message about how to register it.

Reply:
It will hurt trimming but I'm sure we're going to end up with a new API that turns most of these off by default.

Reply:
Not a problem if excluded by empty/slim builder.